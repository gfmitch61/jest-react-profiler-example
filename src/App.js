import React from 'react';
import PropTypes from 'prop-types';

const App = ({ title }) => {
  const [count, setCount] = React.useState(0);

  return (
    <div>
      <h1>{title}</h1>

      <p>{count}</p>

      <button type="button" onClick={() => setCount(count + 1)}>
        Increase
      </button>
    </div>
  );
};

App.propTypes = {
  title: PropTypes.string.isRequired,
};

export default App;
