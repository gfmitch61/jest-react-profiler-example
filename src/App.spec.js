import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer'; // Jest

import { withProfiler } from 'jest-react-profiler';

import App from './App';

const AppWithProfiler = withProfiler(App);

describe('App', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  test('renders App', () => {
    const component = renderer.create(<App title="Hello World" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Doesn't commit ...

  // test('Jest', () => {
  //   renderer.create(<AppWithProfiler title="Hello World" />);

  //   expect(AppWithProfiler).toHaveCommittedTimes(1);
  // });

  test('React Utils', () => {
    act(() => {
      ReactDOM.render(
        <AppWithProfiler title="Hello World" />,
        container
      );
    });

    const paragraph = container.querySelector('p');
    const button = container.querySelector('button');

    expect(paragraph.textContent).toBe('0');
    expect(AppWithProfiler).toHaveCommittedTimes(1);

    act(() => {
      button.dispatchEvent(
        new MouseEvent('click', { bubbles: true })
      );
    });

    expect(paragraph.textContent).toBe('1');
    // expect(AppWithProfiler).toHaveCommittedTimes(2); // doesn't work
  });
});
