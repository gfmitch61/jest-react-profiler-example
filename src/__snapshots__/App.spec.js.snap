// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Increase
  </button>
</div>
`;
